### DISCUSSION TITLE
How do I stop all of the NextJS public directory being precached?

### DISCUSSION THREAD

**Discussion by instaproadrian** [serwist/serwist#158] [Q&A] [answered]
Hi!

I have a NextJS app with several hundred MB of images in the public folder, and obviously I don't want all of them being precached. 

I've tried playing with exclude in `withSerwistInit` but without success, like

```js
const withSerwist = withSerwistInit({
  swSrc: "src/service-worker/sw.ts",
  swDest: "public/sw.js",
  exclude: [
    /\.map$/,
    /^manifest.*\.js$/,
    /\.png$/,
    /\.svg$/,
    /\.gif$/,
    /\.jpg$/,
  ],
})
```

I'm guessing I'm being stupid, or I've missed a key part of the documentation. Can anyone help me?

Thanks

**Comment by DuCanhGH** [comment, marked as answer]
@instaproadrian you should use `globPublicPatterns` :) A bit confusing, that `exclude` option is, but it is not used for public files.

**Reply by instaproadrian** [reply]
Ah, nice! Thanks! 

I think that `globPublicPatterns` isn't currently in the docs, by the way.

**Reply by DuCanhGH** [reply]
@instaproadrian oh, you're right. Must've forgotten :)
